static/js/user_profile.js

```diff
@@ -202,4 +219,15 @@ export function register_click_handlers() {
     $("body").on("click", "#user-profile-modal .stream_list_item", () => {
         hide_user_profile();
     });
+
+    $("body").on("input", "#streams-tab .stream-search", () => {
+        const input = $("#streams-tab .stream-search");
+        if (input.val().trim().length > 0) {
+            $("#streams-tab #clear_stream_search").show();
+            input.css("margin-right", "-20px");
```

Comment:
It's slightly weird to set this `right-margin` css rule based on `x` button visibiltiy. Because of conditional visibility in this case it was hard to have a fixed set of CSS rules and desirable alignment at the same time. I added this css rule here to avoid alignment issues when cross button was displayed like:
![Screenshot_2021-07-01 test - Zulip Dev - Zulip](https://user-images.githubusercontent.com/63504956/124028258-6c3a6e80-da11-11eb-8f03-8bca9a5d27f0.png)

Reply:
Hmm, that definitely looks broken without that.  It may be worth starting a thread in #frontend about making a bit more of a system for these filter boxes that would involve less copy-paste and make that failure mode impossible.

Reply:
As these setting widgets are not so common part of UI, so a simple solution would be to have `x` button present even when there is no text in it.

(I'll start a thread for detailed discussion on this.)

Reply:
Ideally it'd be an option pass into the component whether the `x` should display for an empty text box; I think we have some of each style.

Reply:
Ideally that'd be a flag/option for the widget, "show_close_x_when_empty" that we'd set to false here.